safe/common/parameters/test/example_group_select.py

```diff
@@ -34,7 +32,8 @@
 
 def main():
     """Main function to run the example."""
-    app, iface = qgis_app()
+    from safe.test.utilities import get_qgis_app
+    QGIS_APP, CANVAS, IFACE, PARENT = get_qgis_app(qsetting='InaSAFETest')
```

Comment:
Why do you rollback this change? We are trying to update our test suite to use the internal mock iface provided by QGIS core instead of own one (not very well maintained in InaSAFE)
Especially with the move to QGIS 3 soon, this code will be updated again.

Reply:
because @ismailsunni said that it would be nice if we are using qgis_app in for testing purpose then we set the qsetting to `InaSAFETest` context so it doesn't change the main context. So I assume that I need to update every qgis_app usage with the new one that I just updated. 

Are you suggesting me to leave the usage of the `qgis_app()` as it is?

Reply:
`get_qgis_app` is the legacy way of running tests. We want to move to the builtin QGIS core class instead of InaSAFE class. I tried a prototype with the `qgis_app`. So IMHO, we are moving backward with this change

Reply:
ah yes I remember, we've tried `qgis_app()` before as we updated the code for headless, but it seems like the `qgis_app()` causing a segmentation fault as we developed the Inasafe headless porting to version 4 (CMIIW @ismailsunni @lucernae ). So I'm just being pragmatic to update the code that works for headless now.

Reply:
I will merge this PR first since we urgently need this for headless internationalization. If you still feel uncomfortable with this, I have created a ticket relating to this https://github.com/inasafe/inasafe/issues/4982 . (I'm agree with your point of using qgis built in testing utilities)